rocAL/rocAL/include/meta_data/bounding_box_graph.h

```diff
@@ -25,13 +25,15 @@ THE SOFTWARE.
 #include "meta_data_graph.h"
 #include "meta_node.h"
 
-typedef  struct { double xc; double yc; double w; double h; } BoundingBoxCord_xcycwh;
+typedef  struct { float xc; float yc; float w; float h; } BoundingBoxCord_xcycwh;
+typedef struct { float l; float t; float r; float b; } BoundingBoxCord_ltrb;
+typedef union { BoundingBoxCord_xcycwh xcycwh; BoundingBoxCord_ltrb ltrb; } BoundingBoxCordf;
 
 class BoundingBoxGraph : public MetaDataGraph
 {
 public:
-    void process(pMetaDataBatch meta_data) override;
-    void update_random_bbox_meta_data(pMetaDataBatch meta_data, decoded_image_info decoded_image_info,crop_image_info crop_image_info) override;
-    void update_box_encoder_meta_data(std::vector<float> *anchors, pMetaDataBatch full_batch_meta_data ,float criteria, bool offset , float scale, std::vector<float>& means, std::vector<float>& stds) override;
+    void process(pMetaDataBatch input_meta_data, pMetaDataBatch output_meta_data) override;
+    void update_random_bbox_meta_data(pMetaDataBatch input_meta_data, pMetaDataBatch output_meta_data, decoded_image_info decoded_image_info,crop_image_info crop_image_info) override;
+    void update_box_encoder_meta_data(std::vector<float> *anchors, pMetaDataBatch full_batch_meta_data ,float criteria, bool offset , float scale, std::vector<float>& means, std::vector<float>& stds, float *encoded_boxes_data, int *encoded_labels_data) override;
```

Comment:
Do we need to pass pointers? Can we get reference?

Reply:
Arrays can be passed only as pointers right?